### PR TITLE
Fix Custom Field Flow demos

### PR DESCRIPTION
**Pull request by shadikhani**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-custom-field-flow/52)

### CODE REVIEW

**Review by alvarezguille — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-custom-field-flow/52)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @shadikhani)

___
*[vaadin-custom-field-flow-integration-tests/pom-bower-mode.xml, line 101 at r2](https://reviewable.io/reviews/vaadin/vaadin-custom-field-flow/52#-LuWpLgcEQX7vOmhz1Co:-LuWpLgcEQX7vOmhz1Cp:b-kz3y9g) ([raw file](https://github.com/vaadin/vaadin-custom-field-flow/blob/e9fbb60924a89b2b904599d8fcff7ea5da6732a0/vaadin-custom-field-flow-integration-tests/pom-bower-mode.xml#L101)):*
<details><summary><i alt="summary">Quoted 16 lines of code…</i></summary>

> ```
>         <dependency>
>             <groupId>com.vaadin</groupId>
>             <artifactId>vaadin-date-picker-flow</artifactId>
>             <version>1.3.0</version>
>             <scope>compile</scope>
>         </dependency>
>         <dependency>
>             <groupId>com.vaadin</groupId>
>             <artifactId>vaadin-time-picker-flow</artifactId>
>             <version>1.1-SNAPSHOT</version>
>         </dependency>
>         <dependency>
>             <groupId>com.vaadin</groupId>
>             <artifactId>flow-component-demo-helpers</artifactId>
>             <version>1.5-SNAPSHOT</version>
>         </dependency>
> ```
</details>


same dependencies were added to `vaadin-custom-field-flow-integration-tests/pom.xml` with different versions, same version should be used in both
___
*[vaadin-custom-field-flow-integration-tests/src/main/java/com/vaadin/flow/component/customfield/examples/CustomFieldDemoPage.java, line 20 at r2](https://reviewable.io/reviews/vaadin/vaadin-custom-field-flow/52#-LuWpwAM62J8QKQRHbIT:-LuWpwAM62J8QKQRHbIU:bx96qpb) ([raw file](https://github.com/vaadin/vaadin-custom-field-flow/blob/e9fbb60924a89b2b904599d8fcff7ea5da6732a0/vaadin-custom-field-flow-integration-tests/src/main/java/com/vaadin/flow/component/customfield/examples/CustomFieldDemoPage.java#L20)):*
> ```Java
> @Route("vaadin-custom-field-test-demo")
> public class CustomFieldDemoPage extends DemoView {
> ```

Is `CustomFieldDemoPage` actually used in tests?

File: vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java [hunk 1]
```diff
@@ -1,161 +1,174 @@
...
-        }
+    // source-example-heading: Value change event
+    public static class ValueChangeEvent extends VerticalLayout {
+        private final Div display = new Div();
 
-        @Override
-        protected String generateModelValue() {
-            return wrappedField.getValue();
+        ValueChangeEvent() {
+            // The code is reused from the previous example.
```
Comment (yuriy-fix) [inline vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java:82]: I would probably remove padding as it creates too much free space around the field.

File: vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java [hunk 2]
```diff
@@ -1,161 +1,174 @@
...
-        @Override
-        protected void setPresentationValue(String newPresentationValue) {
-            wrappedField.setValue(newPresentationValue);
+    // begin-source-example
+    // source-example-heading: Required
+    public static class MainForRequired extends VerticalLayout {
+        private final PriceField priceField = new PriceField();
+
+        MainForRequired() {
+            Binder<Product> binder = new Binder<>();
```
Comment (yuriy-fix) [inline vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java:97]: The same thing about indents (padding). Should be removed in order to be consistent with the first demo.

File: vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java [hunk 3]
```diff
@@ -1,161 +1,174 @@
...
+            Product product = new Product();
+            binder.forField(priceField)
+                    .asRequired("Please fill the price amount")
```
Comment (yuriy-fix) [inline vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java:100]: Probably some other validation constraints should be added to the field to show the actual validation apart from the `required` field.

File: vaadin-custom-field-flow-integration-tests/src/main/java/com/vaadin/flow/component/customfield/examples/CustomFieldDemoPage.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+import com.vaadin.flow.router.Route;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.function.Function;
+
+@Route("vaadin-custom-field-test-demo")
+public class CustomFieldDemoPage extends DemoView {
```
Comment (yuriy-fix) [inline vaadin-custom-field-flow-integration-tests/src/main/java/com/vaadin/flow/component/customfield/examples/CustomFieldDemoPage.java]: Do we need it? It's not used in tests.

**Review by yuriy-fix — CHANGES_REQUESTED** [review]
Overall good. 
Some small changes are required

File: vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java (hunk 1 above)
Comment (shadikhani) [reply vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java:82]: Done.

File: vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java (hunk 2 above)
Comment (shadikhani) [reply vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java:97]: Done.

File: vaadin-custom-field-flow-integration-tests/src/main/java/com/vaadin/flow/component/customfield/examples/CustomFieldDemoPage.java (hunk 4 above)
Comment (shadikhani) [reply vaadin-custom-field-flow-integration-tests/src/main/java/com/vaadin/flow/component/customfield/examples/CustomFieldDemoPage.java]: Done. I removed the CustomFieldDemoPage.java as it has nothing to do with the test.

**Review by shadikhani — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-custom-field-flow/52)* status: 0 of 4 files reviewed, 6 unresolved discussions (waiting on @alvarezguille, @shadikhani, and @yuriy-fix)

___
*[vaadin-custom-field-flow-integration-tests/pom-bower-mode.xml, line 101 at r2](https://reviewable.io/reviews/vaadin/vaadin-custom-field-flow/52#-LuWpLgcEQX7vOmhz1Co:-Lubn74T-N9tpHWbjW4y:bj6ij4t) ([raw file](https://github.com/vaadin/vaadin-custom-field-flow/blob/e9fbb60924a89b2b904599d8fcff7ea5da6732a0/vaadin-custom-field-flow-integration-tests/pom-bower-mode.xml#L101)):*
<details><summary><i>Previously, alvarezguille (Guille) wrote…</i></summary><blockquote>

> ```
>         <dependency>
>             <groupId>com.vaadin</groupId>
>             <artifactId>vaadin-date-picker-flow</artifactId>
>             <version>1.3.0</version>
>             <scope>compile</scope>
>         </dependency>
>         <dependency>
>             <groupId>com.vaadin</groupId>
>             <artifactId>vaadin-time-picker-flow</artifactId>
>             <version>1.1-SNAPSHOT</version>
>         </dependency>
>         <dependency>
>             <groupId>com.vaadin</groupId>
>             <artifactId>flow-component-demo-helpers</artifactId>
>             <version>1.5-SNAPSHOT</version>
>         </dependency>
> ```

same dependencies were added to `vaadin-custom-field-flow-integration-tests/pom.xml` with different versions, same version should be used in both
</blockquote></details>

 I removed the dependencies, I also removed the CustomFieldDemoPage.java. The old demos have nothing to do with the tests.
___
*[vaadin-custom-field-flow-integration-tests/src/main/java/com/vaadin/flow/component/customfield/examples/CustomFieldDemoPage.java, line 20 at r2](https://reviewable.io/reviews/vaadin/vaadin-custom-field-flow/52#-LuWpwAM62J8QKQRHbIT:-LuboGV61uprpI6WYTdI:b4ghpdz) ([raw file](https://github.com/vaadin/vaadin-custom-field-flow/blob/e9fbb60924a89b2b904599d8fcff7ea5da6732a0/vaadin-custom-field-flow-integration-tests/src/main/java/com/vaadin/flow/component/customfield/examples/CustomFieldDemoPage.java#L20)):*
<details><summary><i>Previously, alvarezguille (Guille) wrote…</i></summary><blockquote>

Is `CustomFieldDemoPage` actually used in tests?
</blockquote></details>


I removed the dependencies, I also removed the CustomFieldDemoPage.java. The old demos have nothing to do with the tests.

**Review by alvarezguille — COMMENTED** [review]
> Reviewed 2 of 3 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-custom-field-flow/52)* status: 1 of 3 files reviewed, 4 unresolved discussions (waiting on @shadikhani and @yuriy-fix)

File: vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java (hunk 3 above)
Comment (yuriy-fix) [reply vaadin-custom-field-flow-demo/src/main/java/com/vaadin/flow/component/customfield/vaadincom/CustomFieldView.java:100]: Based on 1:1 discussion this probably should be changed for all of the components demos at once. Not in scope of this PR

**Review by yuriy-fix — APPROVED** [review]

**Review by alvarezguille — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, 1 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-custom-field-flow/52#-:-Lufz3zVB0BVwWU8zVsk:bnfp4nl)* status: all files reviewed, 4 unresolved discussions (waiting on @shadikhani and @yuriy-fix)

**Review by yuriy-fix — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, 3 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/vaadin/vaadin-custom-field-flow/52#-:-Lug0cLf49xTSRnjwa7d:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**